app/src/main/java/com/github/se/eduverse/ui/navigation/TopNavigationBar.kt

```diff
@@ -1,38 +1,62 @@
 package com.github.se.eduverse.ui.navigation
 
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.size
 import androidx.compose.material.icons.Icons
 import androidx.compose.material.icons.automirrored.filled.ArrowBack
 import androidx.compose.material3.ExperimentalMaterial3Api
 import androidx.compose.material3.Icon
 import androidx.compose.material3.IconButton
 import androidx.compose.material3.MaterialTheme
-import androidx.compose.material3.Text
 import androidx.compose.material3.TopAppBar
 import androidx.compose.material3.TopAppBarDefaults
 import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import com.github.se.eduverse.R
+import com.github.se.eduverse.ui.theme.md_theme_light_onTertiary
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun TopNavigationBar(
-    screenTitle: String,
     navigationActions: NavigationActions,
-    actions: @Composable (RowScope.() -> Unit) = {}
+    actions: @Composable() (RowScope.() -> Unit) = {}
 ) {
   TopAppBar(
-      modifier = Modifier.testTag("topNavigationBar"),
-      title = { Text(screenTitle, Modifier.testTag("screenTitle")) },
+      modifier =
+          Modifier.fillMaxWidth()
+              .background(
+                  Brush.horizontalGradient(
+                      colors =
+                          listOf(
+                              MaterialTheme.colorScheme.secondary,
+                              MaterialTheme.colorScheme.primary))),
+      title = {
+        Box(modifier = Modifier.fillMaxWidth(), contentAlignment = Alignment.Center) {
+          Image(
+              painter = painterResource(id = R.drawable.eduverse_logo_png),
+              contentDescription = "Logo",
+              modifier = Modifier.size(140.dp).testTag("centerImage"))
```

Comment:
For most screens I think it is ok to replace the title of the screen by the app logo, however for some of them it was used to display important information (typicaly for some folder-related screens, the title of the TopNavigationBar was used to indicated the name of the opened folder). 

In order to fix this, you could add an optional argument to TopNavigationBar. Then if it is not null display the text otherwise show the logo.

```
fun TopNavigationBar(
    screenTitle: String,
    navigationActions: NavigationActions,
    screenTitle: String? = null,
    actions: @Composable (RowScope.() -> Unit) = {}
) {
  ...
    title = {
      if (title == null) {
        Text(title)
      } else {
        ...
```